clients/src/main/java/org/apache/kafka/common/protocol/types/Type.java

```diff
@@ -184,14 +184,19 @@ public Long validate(Object item) {
         public void write(ByteBuffer buffer, Object o) {
             byte[] bytes = Utils.utf8((String) o);
             if (bytes.length > Short.MAX_VALUE)
-                throw new SchemaException("String is longer than the maximum string length.");
+                throw new SchemaException("String length " + bytes.length + " is larger than the maximum string length.");
             buffer.putShort((short) bytes.length);
             buffer.put(bytes);
         }
 
         @Override
         public Object read(ByteBuffer buffer) {
-            int length = buffer.getShort();
+            short length = buffer.getShort();
+            if (length < 0)
+                throw new SchemaException("String length " + length + " cannot be negative");
+            if (length > buffer.remaining())
```

Comment:
Yes, if we want to completely eliminate those, then that would be needed. But it would not be enough either, we would have to check before we get a user-supplied size/length too for example. I don't feel too strongly, so I can change it if others also think I should, but I am leaning towards leaving as is. I am a bit concerned that increasing bytecode size for the simple methods could also have an impact on inlining and ultimately performance.